### PR TITLE
Add support for private hub

### PR DESCRIPTION
**Pull request by juliensimon**
This fixes https://github.com/huggingface/huggingface_hub/issues/650

We simply set HF_ENDPOINT to the URL of a private hub, and use the API as-is.  If the API itself has the endpoint, then the environment variable is ignored.

FYI, the same variable is used by the datasets library for a similar purpose.

### CODE REVIEW

File: src/huggingface_hub/hf_api.py [hunk 1]
```diff
@@ -410,6 +410,8 @@ def erase_from_credential_store(username=None):
 
 class HfApi:
     def __init__(self, endpoint=None):
+        if endpoint is None:
+            endpoint = os.getenv('HF_ENDPOINT')
         self.endpoint = endpoint if endpoint is not None else ENDPOINT
```
Comment (julien-c) [inline src/huggingface_hub/hf_api.py:415]: maybe more Pythonic to do those three lines in one line
```python
self.endpoint = endpoint if endpoint is not None else os.getenv('HF_ENDPOINT', ENDPOINT)
```

Comment (julien-c) [reply src/huggingface_hub/hf_api.py:415]: other than that LGTM=)

Comment (juliensimon) [reply src/huggingface_hub/hf_api.py]: ok!

**Review by julien-c — APPROVED** [review]

**Review by osanseviero — APPROVED** [review]
Looks good!

**Review by LysandreJik — APPROVED** [review]
Yes, LGTM! Thank you @juliensimon

**Comment by juliensimon** [comment]
Thanks!

**Pull request merged**